src/lib/core/diag.h

```diff
@@ -120,6 +124,72 @@ error_ref(struct error *e);
 void
 error_unref(struct error *e);
 
+static inline const char *
+error_field_get_str(const struct error *e, const char *name)
```

Comment:
There's struct error_field so I'd expect error_field_* to take it, but it takes struct error instead. Maybe, rename these functions to `error_{get,set}_{int,double,str}_field` or even simply `error_{get,set}_{int,double,str}`?

I'd also rename `error_field_find` (added by a following patch) to `error_find_field`.

Reply:
Totally agree, thanks for the suggestions. Also, what do you think if we would rename `error_payload` to `error_dict`? It is shorter. I don't like how long `payload` name is. Or even `error_kv`.

Diff is scattered over several commits, so I won't attach it here. I used your proposal about `error_{get,set}_{type}`.

Reply:
Thanks. Re `error_payload` - I like the name; `error_dict` or `error_kv` sound like a dictionary/kv-storage of errors, which would be confusing.